app/src/main/java/es/wolfi/app/passman/CredentialEdit.java

```diff
@@ -222,7 +254,32 @@ public void run() {
         return new View.OnClickListener() {
             @Override
             public void onClick(View view) {
-                ((PasswordList) Objects.requireNonNull(getActivity())).selectFileToAdd();
+                ((PasswordList) Objects.requireNonNull(getActivity())).selectFileToAdd(2);
+            }
+        };
+    }
+
+    public View.OnClickListener getAddCustomFieldButtonListener() {
+        return new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if (customFieldType.getSelectedItem().toString().equals("File")) {
+                    ((PasswordList) Objects.requireNonNull(getActivity())).selectFileToAdd(3);
+                } else {
+                    try {
+                        JSONObject customFieldJSONObject = new JSONObject();
```

Comment:
I believe this kind of data handling would be better placed on the [api library](https://github.com/nextcloud/passman-android/tree/master/app/src/main/java/es/wolfi/passman/API)

Reply:
fixed in 5b250f8317c18764586a3d11ea130c0c944160c7